core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -768,6 +754,23 @@ private object RTS {
 
     final def join: IO[E, A] = IO.async0(join0)
 
+    final override def finished(f: ExitResult[E, A] => IO[Nothing, Unit]): IO[Nothing, Unit] =
+      IO.sync(finished0(f))
+
+    @tailrec
+    final def finished0(f: ExitResult[E, A] => IO[Nothing, Unit]): Unit = {
+      val oldStatus = status.get
+
+      oldStatus match {
+        case x @ Executing(_, _, _, _, _) =>
```

Comment:
is `case x: Executing =>` not sufficient?

Reply:
Yes 😃 

I don't know if there is a performance impact of using `.copy` rather than pattern matching and creating a new object: haven't found anything online, but the former is used everywhere else in `RTS` which makes me think there is. In which case I'm happy to switch!